app/services/workflows/non_passported_workflow.rb

```diff
@@ -1,37 +1,39 @@
-                                  proceeding_types: assessment.proceeding_types,
-                                  receives_qualifying_benefit: applicant.details.receives_qualifying_benefit,
-                                  receives_asylum_support: applicant.details.receives_asylum_support)
-          end
-        end
+        calculation_output = if gross_income_subtotals.gross.ineligible?
+                               CalculationOutput.new(gross_income_subtotals: gross_income_subtotals.gross,
+                                                     proceeding_types: assessment.proceeding_types,
+                                                     receives_qualifying_benefit: applicant.details.receives_qualifying_benefit,
+                                                     receives_asylum_support: applicant.details.receives_asylum_support,
+                                                     disposable_income_subtotals: unassessed_disposable_income(assessment:),
+                                                     capital_subtotals: unassessed_capital)
+                             else
+                               disposable_income_subtotals = get_disposable_income_subtotals(assessment:, applicant:, partner:, gross_income_subtotals: gross_income_subtotals.gross)
+                               if disposable_income_subtotals.ineligible?
+                                 CalculationOutput.new(gross_income_subtotals: gross_income_subtotals.gross,
+                                                       disposable_income_subtotals:,
+                                                       capital_subtotals: unassessed_capital,
+                                                       proceeding_types: assessment.proceeding_types,
+                                                       receives_qualifying_benefit: applicant.details.receives_qualifying_benefit,
+                                                       receives_asylum_support: applicant.details.receives_asylum_support)
+                               else
+                                 capital_subtotals = get_capital_subtotals(assessment:, applicant:, partner:, disposable_income_subtotals:)
+                                 CalculationOutput.new(gross_income_subtotals: gross_income_subtotals.gross,
+                                                       disposable_income_subtotals:, capital_subtotals:,
+                                                       proceeding_types: assessment.proceeding_types,
+                                                       receives_qualifying_benefit: applicant.details.receives_qualifying_benefit,
+                                                       receives_asylum_support: applicant.details.receives_asylum_support)
+                               end
+                             end
+
+        Result.new calculation_output:, remarks: gross_income_subtotals.remarks
```

Comment:
@starswan What do you think of returning similar Result object for Passported workflow?

Reply:
I thought of that, but it would always be empty.